lib/index.js

```diff
@@ -112,8 +159,49 @@ MaplibreGeocoder.prototype = {
         "</div></div>"
       );
     },
+    processResults: function (searchResults, suggestionResults, limit) {
+      if (!suggestionResults || suggestionResults.length === 0) {
+        return searchResults;
+      }
+
+      // Filter out the suggestions that are already in the search results
+      const featureLabels = searchResults.map(function (feature) {
+        return feature.place_name;
+      });
+      const filteredSuggestions = suggestionResults.filter(function (result) {
+        return !featureLabels.includes(result);
+      });
+
+      if (filteredSuggestions.length + searchResults.length >= limit) {
+        // Restrict suggestions to < half of the limit at most
+        let suggestionsLimit = Math.min(
+          filteredSuggestions.length,
+          Math.floor(limit / 2)
+        );
+        // If there are not enough results to fill to limit, add more suggestions
+        while (searchResults.length + suggestionsLimit < limit) {
+          suggestionsLimit++;
```

Comment:
If an absurd number for limit is passed, won't it cause out of index on suggestionsLimit?

Reply:
Sorry I missed that. Wouldn't we want the condition to be only `> limit` and not `>= limit` to apply further filtering?

Also is `limit` a required options param? I couldn't find the documentation of the `processResults` function.

Reply:
Good point on the `>=` vs `>`, it wasn't going to cause a bug, but it was not needed.

The `limit` comes from the `options` object of the Geocoder. You can see where `processResults` is called on line [776](https://github.com/maplibre/maplibre-gl-geocoder/pull/27/files#diff-92bbac9a308cd5fcf9db165841f2d90ce981baddcb2b1e26cfff170929af3bd1R776).

Yes, the `processResults` function needs proper documentation still, that's coming today.